DatadogCrashReporting/Sources/Integrations/CrashReportSender.swift

```diff
@@ -39,30 +46,29 @@ internal struct MessageBusSender: CrashReportSender {
     ///   - report: The crash report.
     ///   - context: The crash context
     func send(report: DDCrashReport, with context: CrashContext) {
-        guard context.trackingConsent == .granted else {
+        guard let core = core, context.trackingConsent == .granted else {
             return
         }
 
-        sendCrash(
-            baggage: [
-                "report": report,
-                "context": context
-            ]
-        )
-    }
-
-    private func sendCrash(baggage: FeatureBaggage) {
-        core?.send(
-            message: .custom(key: MessageKeys.crash, baggage: baggage),
-            else: {
-                DD.logger.warn(
-            """
-            In order to use Crash Reporting, RUM or Logging feature must be enabled.
-            Make sure `.enableRUM(true)` or `.enableLogging(true)` are configured
-            when initializing Datadog SDK.
-            """
+        do {
+            try core.send(
+                message: .baggage(
+                    key: MessageKeys.crash,
+                    value: Crash(report: report, context: context)
+                ),
+                else: {
+                    DD.logger.warn(
+                """
+                In order to use Crash Reporting, RUM or Logging feature must be enabled.
+                Make sure `.enableRUM(true)` or `.enableLogging(true)` are configured
+                when initializing Datadog SDK.
+                """
+                )
+                }
             )
-            }
-        )
+        } catch {
+            core.telemetry
+                .error("Fails to encode crash", error: error)
+        }
```

Comment:
suggestion/ It feels that throwing with `try core.send(...)` is not a productive choice:
* baggage encoding can fail, so we need `try {}`,
* there is almost never a fallback we could implement in `catch {}`,
* we may always want to know the failure in telemetry (all these failures are critical).

All that together, it would be more productive to offer non-throwing `core.send(message: .baggage(...))` that will report every error to telemetry. To differentiate one failure from the other, the `send(message:)` could accept [`#fileID`](https://developer.apple.com/documentation/swift/fileid()) and `#line` as default parameters. More context about the error could be also inferred from the `baggage.key`.

In this PR, we introduce the same pattern in multiple places - all could be simplified if we go this way. Also, in `RUMSessionScope` we opt-out from capturing errors with `try?` - that would be automatically covered with this proposal. Same for next usages - we will get telemetry for free.

LMKWDYT @maxep

Reply:
That's a good point, thanks! It's indeed not very practical to catch the error.
Using `#fileID` and `#line` is a good idea, but I'm not sure how it can be applied since it's [the `FeatureBaggage` method](https://github.com/DataDog/dd-sdk-ios/blob/7981a6d45cebd13a5296ef2ba65dad9190d94b03/DatadogInternal/Sources/MessageBus/FeatureMessage.swift#L37) that throw, not the core. I will do some test!

Reply:
So I went another path which is more efficient and more convenient:
The `FeatureBaggage` will delay the encoding to when the message is actually received. Meaning that:
1. We won't encode the value if it's never read
2. We will throw an error only when decoding the value of the baggage

This way, we don't need to `try core.send(message: .baggage(...))`, the `FeatureMessageReceiver` will be responsible for reporting telemetry on baggage decoding failure.